packages/data/src/components/registry-provider.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext, createHigherOrderComponent, createElement } from '@wordpress/element';
+
+const { Consumer, Provider } = createContext( null );
+
+export { Provider as RegistryProvider };
+
+/**
+ * A Higher Order Component used to inject the data registry as a prop
+ * to the wrapped component.
+ *
+ * @param {Componoent} OriginalComponent Component to wrap.
+ *
+ * @return {Component} Enhanced component with injected context as props.
+ */
+export const withRegistry = createHigherOrderComponent( ( OriginalComponent ) => {
+	return ( props ) => (
+		<Consumer>
+			{ ( registry ) => (
+				<OriginalComponent registry={ registry } { ...props } />
```

Comment:
We'll have the same problem I encountered in #7453, which is that `withDispatch` and `withSelect` pass through all props they receive. And for some components this is problematic because they then spread onto the DOM elements (e.g. `Button`), thus producing warnings about unrecognized attributes.

To preempt the obvious solution to use `omit( this.props, [ 'registry' ] )`, I prefer my solution in #7453 which is marginally more performant: pass the props unaffected by a separate `ownProps` prop. This also has the effect of not including `registry` in `mapSelectToProps` / `mapDispatchToProps`.